wallet/src/main/java/com/android/identity_credential/wallet/PresentationActivity.kt

```diff
@@ -379,18 +330,27 @@ class PresentationActivity : FragmentActivity() {
         if (state.value == State.REQUEST_AVAILABLE) {
             val deviceResponseGenerator =
                 DeviceResponseGenerator(Constants.DEVICE_RESPONSE_STATUS_GENERAL_ERROR)
-            transferHelper?.sendResponse(deviceResponseGenerator.generate())
+            sendResponseToDevice(deviceResponseGenerator.generate())
         }
         deviceRetrievalHelper?.disconnect()
         deviceRetrievalHelper = null
         transport = null
         handover = null
-        state.value = State.NOT_CONNECTED
+        finish()
     }
 
-    private fun getDeviceRequest(): ByteArray {
-        check(state.value == State.REQUEST_AVAILABLE) { "Not in REQUEST_AVAILABLE state" }
-        check(deviceRequest != null) { "No request available " }
-        return deviceRequest as ByteArray
-    }
-}
+    /**
+     * Send response bytes to requesting party and updates state to [State.RESPONSE_SENT]
+     * @param deviceResponseBytes response bytes that may or may not have been processed (such as
+     * when sending an error)
+     */
+    private fun sendResponseToDevice(deviceResponseBytes: ByteArray) =
+        deviceRetrievalHelper?.run {
+            sendDeviceResponse(
+                deviceResponseBytes,
+                Constants.SESSION_DATA_STATUS_SESSION_TERMINATION
+            )
+            state.value = State.RESPONSE_SENT
+        }
+            ?: throw IllegalStateException("Unable to send response bytes, deviceRetrievalHelper is [null].")
+}
```

Comment:
This seems unnecessary to have in a helper function, just use deviceRetrievalHelper from the call site.

Reply:
Totally! that makes sense however this is a reusable function that is called in 2 places, it hides the low level details of what it takes to send bytes via `deviceRetrievalHelper`. Copy pasting that code around doesn't sound too maintainable and it costs 7 lines of code vs 1 line to call "sendResponseToDevice".

I think the current implementation clear to read, it's 1 line of code
```
            sendResponseToDevice(bytes)
```
  
  
Without the function `sendResponseToDevice(bytes)`, the `deviceRetrievalHelper` code can be inlined to

```
       deviceRetrievalHelper?.run {
            sendDeviceResponse(
                deviceResponseGenerator.generate(),
                Constants.SESSION_DATA_STATUS_SESSION_TERMINATION
            )
            state.value = State.RESPONSE_SENT
        }
```

Ends up costing the reviewer 14 lines to read vs 2 + 7 lines